### ISSUE TITLE
Deprecate sep="\n"?

### ISSUE THREAD

**Issue by MichaelChirico** [Rdatatable/data.table#5839]
https://github.com/Rdatatable/data.table/blob/6c1fd839e0e0257d13f07975ead0fe6fdfee2f61/NEWS.1.md?plain=1#L1209

This NEWS item (1.11.0, 2018) mentions `sep="\n"` would be deprecated, but I don't know why. It seems to work fine:

```r
fread("a\nb\nc", sep = "\n")
#    a
# 1: b
# 2: c
```

And here `sep` is set to `\n` before sending on to C:

https://github.com/Rdatatable/data.table/blob/6c1fd839e0e0257d13f07975ead0fe6fdfee2f61/R/fread.R#L12-L15

I guess it's about clarity for Windows, where `sep="\n"` doesn't mean literally `\n`, but "end-of-line", i.e., `"\n"` is actually a "sentinel".

But it doesn't seem to have caused any issue in the interim. Do we want to follow through with deprecating `sep = "\n"`?

**Comment by jangorecki** [comment]
If we don't know why to deprecate we can leave it as is, till someone come and ask for that. It feels it can be still useful.

**Comment by MichaelChirico** [comment]
I agree. Leaving open for now to get more opinions. If we leave things be, we should add a NEWS note about the change of plans.
